testsuite/diem-swarm/src/swarm.rs

```diff
@@ -23,49 +23,60 @@ use std::{
 use thiserror::Error;
 
 pub struct DiemNode {
-    node: Child,
-    node_id: String,
-    validator_peer_id: Option<AccountAddress>,
-    role: RoleType,
+    process: Child,
+    pub(crate) node_id: String,
+    node_type: NodeType,
+    peer_id: AccountAddress,
     debug_client: NodeDebugClient,
-    port: u16,
-    pub log: PathBuf,
+    pub log_path: PathBuf,
+    config: NodeConfig,
```

Comment:
Would it. be possible to have accessors instead of making some of these fields public? specifically the node_id since that's what you've made public here.

Reply:
Yes, we can totally!

I guess this was a little scrappy / worried more about getting it working than cleaning it up (since the framework already had a mixed bag of both)